src/main/java/com/labsynch/labseer/chemclasses/rdkit/ChemStructureServiceRDKitImpl.java

```diff
@@ -291,23 +406,50 @@ public double getExactMass(String molStructure) throws CmpdRegMolFormatException
 
 	@Override
 	public boolean deleteStructure(String structureTable, int cdId) {
-		// TODO Auto-generated method stub
-		return false;
+		Long id = new Long(cdId);
+		if (structureTable.equalsIgnoreCase("Parent_Structure")){
+			RDKitStructure rdkitStructure = RDKitStructure.findRDKitStructure(id);
+			rdkitStructure.remove();
+		} else if (structureTable.equalsIgnoreCase("SaltForm_Structure")){
+			RDKitSaltFormStructure rdkitSaltFormStructure = RDKitSaltFormStructure.findRDKitSaltFormStructure(id);
+			rdkitSaltFormStructure.remove();
+		} else if (structureTable.equalsIgnoreCase("Salt_Structure")){
+			RDKitSaltStructure rdkitSaltStructure = RDKitSaltStructure.findRDKitSaltStructure(id);
+			rdkitSaltStructure.remove();
+		} else if (structureTable.equalsIgnoreCase("Dry_Run_Compound_Structure")){
+			RDKitDryRunStructure rdkitDryRunStructure = RDKitDryRunStructure.findRDKitDryRunStructure(id);
+			rdkitDryRunStructure.remove();
+		}
+		return true;
 	}
 
 	@Override
 	public CmpdRegMolecule[] searchMols(String molfile, String structureTable, int[] inputCdIdHitList,
 			String plainTable, String searchType, Float simlarityPercent, int maxResults)
 			throws CmpdRegMolFormatException {
-		// TODO Auto-generated method stub
-		return null;
+		
+			RDKitStructure serviceRDKitStructure = getRDKitStructureFromService(molfile);
+
+			//Create empty list
+			List<RDKitStructure> rdkitStructures = new ArrayList<RDKitStructure>();
+			if (searchType.equalsIgnoreCase("FULL_TAUTOMER")){
```

Comment:
Is there a reason both this and `searchMolStructures` contain search implementations? Not inherently a problem, just looking at how many different `search` functions there are, and wondering if we can refactor it down to only one function that actually does searching, and keep the rest as wrappers.

I do see that this and `searchMolStructures` return different result types: `CmpdRegMolecule[]` vs `int[]` but it still seems that could be wrapped.

Reply:
I think this makes sense to refactor now. I will do that.

Reply:
Just merged the refactor.